core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelRangeGrtThanFiterExecuterImpl.java

```diff
@@ -387,9 +387,11 @@ private BitSet getFilteredIndexes(DimensionColumnPage dimensionColumnPage,
     } else {
       bitSet = setFilterdIndexToBitSet(dimensionColumnPage, numerOfRows);
     }
-    if (dimensionColumnPage.isNoDicitionaryColumn()) {
+    if (dimensionColumnPage.isNoDicitionaryColumn() && !dimensionColumnPage.isAdaptiveEncoded()) {
```

Comment:
no need add condition here. Please handle inside 'removeNullValues' method. Not required to do changes in these classes